doc/developers/advanced_installation.rst

```diff
@@ -1,9 +1,21 @@
 
 .. _advanced-installation:
 
-==================================================================
-Installing the development version of scikit-learn (master branch)
-==================================================================
+==================================================
+Installing the development version of scikit-learn
+==================================================
+
+This section introduces how to install the **master branch** of scikit-learn.
+This is required:
+
+- to work on a contribution (bug fix, new feature, code or documentation
+  improvement);
+
+- to try a new feature that will be shipped in the next release (that is, a
+  feature from a pull-request that was recently merged to the master branch);
```

Comment:
We do have nightly wheels for master, so installing from source is probably not the easiest way to go. Maybe change to "build features from pull-requests"?

Reply:
This this the header for "Installing the development version of scikit-learn" and the first subsection is "Installing nightly builds".

Reply:
I pushed a new commit to split this list in the 2 sub-sections.

Reply:
Indeed, didn't see the subsection as it was collapsed.. nevermind.